stdlib/SparseArrays/docs/src/AbstractSparseMatrixCSC.md

```diff
@@ -0,0 +1,62 @@
+    AbstractSparseMatrixCSC{Tv,Ti<:Integer} <: AbstractSparseMatrix{Tv,Ti}
+
+Supertype for matrix with compressed sparse column (CSC).
+
+!!! compat "Julia 1.4"
+    `AbstractSparseMatrixCSC` requires at least Julia 1.4.
+
+## `AbstractSparseMatrixCSC` interface
+
+In addition to the [Abstract array interface](@ref man-interface-array), every
+`AbstractSparseMatrixCSC` subtype `TS` must provide the following methods:
+
+* [`size(::TS)`](@ref size)
+* [`getcolptr(::TS) :: AbstractVector{<:Ti}`](@ref getcolptr)
+* [`rowvals(::TS) :: AbstractVector{<:Ti}`](@ref rowvals)
```

Comment:
In #25118 @simonbyrne suggested `rowindices`.  It is a better name but `rowvals` is already [a documented public API](https://docs.julialang.org/en/latest/stdlib/SparseArrays/#SparseArrays.rowvals) so I don't know if we want to change this at this point.  But this may be a good opportunity to do so?

Reply:
Again agree.

Reply:
In principle, I think we can define `rowvals` to be an alias to `rowindices` like this:

```julia
rowindices(S::SparseMatrixCSC) = getfield(S, :rowval)
rowvals(x) = rowindices(x)
```

and then use `rowindices` internally everywhere.

In the documentation, we can mention that the use of `rowvals` is highly discouraged and will be deprecated in the next major release.

This means that for _end-users_ `rowvals` keep working as-is but for _package authors_ of custom array types they are forced to use `rowindices` from the get-go.